pkg/apiserver/experiment/experiment.go

```diff
@@ -717,3 +699,244 @@ func (s *Service) patchExperiment(exp *ExperimentBase, annotations map[string]st
 		chaosKind.Chaos,
 		client.ConstantPatch(types.MergePatchType, mergePatch))
 }
+
+// @Summary Update the chaos experiment by API
+// @Description Update the chaos experiment by API
+// @Tags experiments
+// @Produce json
+// @Param request body ExperimentInfo true "Request body"
+// @Success 200 "update ok"
+// @Failure 400 {object} utils.APIError
+// @Failure 500 {object} utils.APIError
+// @Router /api/experiments/update [put]
+func (s *Service) updateExperiment(c *gin.Context) {
+	exp := &ExperimentInfo{}
+	if err := c.ShouldBindJSON(exp); err != nil {
+		c.Status(http.StatusBadRequest)
+		_ = c.Error(utils.ErrInvalidRequest.WrapWithNoMessage(err))
+		return
+	}
+
+	updateFuncs := map[string]actionFunc{
+		v1alpha1.KindPodChaos:     s.updatePodChaos,
+		v1alpha1.KindNetworkChaos: s.updateNetworkChaos,
+		v1alpha1.KindIOChaos:      s.updateIOChaos,
+		v1alpha1.KindStressChaos:  s.updateStressChaos,
+		v1alpha1.KindTimeChaos:    s.updateTimeChaos,
+		v1alpha1.KindKernelChaos:  s.updateKernelChaos,
+	}
+
+	f, ok := updateFuncs[exp.Target.Kind]
+	if !ok {
+		c.Status(http.StatusBadRequest)
+		_ = c.Error(utils.ErrInvalidRequest.New("Target kind is not available"))
```

Comment:
How about changing to "_ = c.Error(utils.ErrInvalidRequest.New(exp.Target.Kind + " is not supported")) "?